src/DynamoCore/Logging/AnalyticsService.cs

```diff
@@ -11,6 +13,20 @@ namespace Dynamo.Logging
     /// </summary>
     class AnalyticsService
     {
+        private static readonly ManualResetEventSlim waitInit = new ManualResetEventSlim(false);
+        static AnalyticsService()
+        {
+            // Initialize the concrete class only when we initialize the Service.
+            // This will also load the Analytics.Net.ADP assembly
+            // We must initialize the ADPAnalyticsUI instance before the Analytics.Start call.
+            Task.Run(() =>
+            {
+                adpAnalyticsUI = new ADPAnalyticsUI();
+                Analytics.Start(new DynamoAnalyticsClient());
+                waitInit.Set();
```

Comment:
can you explain or diagram somewhere why we need the ManualResetEvent?

Reply:
We only need to wait for adp initialization to complete and we'll wait for this in all tracking requests.